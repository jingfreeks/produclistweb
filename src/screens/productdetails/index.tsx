export {default as ProductDetails} from './productdetails'