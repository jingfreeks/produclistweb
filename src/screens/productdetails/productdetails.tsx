import React from "react";
import { useGetProdDetailsQuery } from "../../slice/producSlice";
import { useParams } from "react-router-dom";
import {
  LoadingContainerStyled,
  Container,
  ImgStyled,
  TitleStyled,
  InfoContainerStyled,
  DescriptIOnStyled,
} from "./styles";
function Productdetails() {
  const { prodId } = useParams();
  const { data: product, isLoading } = useGetProdDetailsQuery<{
    refetch: () => void;
    data: {image?:string;title?:string;category?:string;price?:string,_id:string;decription:string};
    isLoading: boolean;
    isSuccess: string;
    isError: boolean;
    error: any;
  }>({ prodId: prodId });

  const Loading = () => {
    return (
      <LoadingContainerStyled>
        <h2>ðŸŒ€ Loading...</h2>
      </LoadingContainerStyled>
    );
  };
  if (isLoading) {
    return <Loading />;
  }
  return (
    <Container>
      <InfoContainerStyled>
        <ImgStyled src={product?.image} alt="new" />
        <TitleStyled>{product?.title}</TitleStyled>
      </InfoContainerStyled>
      <InfoContainerStyled>
        <DescriptIOnStyled>{product?.decription}</DescriptIOnStyled>
        <DescriptIOnStyled>Category:{product?.category}</DescriptIOnStyled>
        <DescriptIOnStyled>Price:{product?.price}</DescriptIOnStyled>
      </InfoContainerStyled>
    </Container>
  );
}

export default Productdetails;
