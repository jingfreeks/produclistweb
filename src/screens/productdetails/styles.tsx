import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items:center;
`;
export const LoadingContainerStyled = styled(Container)`
  display: flex;
  justify-content: center;
  height: 100vh;
`;

export const ImgStyled = styled.img`
  width: 250px;
  height: 300px;
  margin-top:20px;
`;

export const TitleStyled = styled.h1`
  font-size: 1.5em;
 text-align: center;
  margin-bottom: -5px;
`;

export const InfoContainerStyled = styled.div`
  display: flex;
  flex-direction: column;
`;

export const DescriptIOnStyled = styled.p`
  font-size: 0.9em;;
`;