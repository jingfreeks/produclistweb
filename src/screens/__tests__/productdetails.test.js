import { renderWithProviders } from "../../slice/testframework";
import {ProductDetails} from "../productdetails";


// Mock useRouter:
describe("Product Details ", () => {
  it("renders Layout unchanged to match snapshot", () => {
    const { container } = renderWithProviders(<ProductDetails />);
    expect(container).toMatchSnapshot();
  });
});
