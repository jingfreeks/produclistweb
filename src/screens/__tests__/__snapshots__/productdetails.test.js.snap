// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Details  renders Layout unchanged to match snapshot 1`] = `
<div>
  <div
    class="sc-blHHSb sc-gtLWhw PDObS irshlR"
  >
    <h2>
      ðŸŒ€ Loading...
    </h2>
  </div>
</div>
`;
